src/plugins/gui/providers/moonpayProvider.ts

```diff
@@ -714,186 +721,156 @@ export const moonpayProvider: FiatProviderFactory = {
-
-                        const message =
-                          sprintf(
-                            lstrings.fiat_plugin_sell_complete_message_s,
-                            cryptoAmount,
-                            displayCurrencyCode,
-                            fiatAmount,
-                            displayFiatCurrencyCode,
-                            '1'
-                          ) +
-                          '\n\n' +
-                          sprintf(
-                            lstrings.fiat_plugin_sell_complete_message_2_hour_s,
-                            '1'
-                          ) +
-                          '\n\n' +
-                          lstrings.fiat_plugin_sell_complete_message_3
-                        await showUi.buttonModal({
-                          buttons: {
-                            ok: { label: lstrings.string_ok, type: 'primary' }
-                          },
-                          title: lstrings.fiat_plugin_sell_complete_title,
-                          message
-                        })
-                      } catch (e: unknown) {
-                        await showUi.exitScene()
-                        // Reopen the webivew on the Paybis payment screen
-                        await openWebView()
-                        if (
-                          e instanceof Error &&
-                          e.message === SendErrorNoTransaction
-                        ) {
-                          await showUi.showToast(
-                            lstrings.fiat_plugin_sell_failed_to_send_try_again,
-                            NOT_SUCCESS_TOAST_HIDE_MS
-                          )
-                        } else if (
-                          e instanceof Error &&
-                          e.message === SendErrorBackPressed
-                        ) {
-                          // Do nothing
-                        } else {
-                          await showUi.showError(e)
-                        }
-                      } finally {
-                        inPayment = false
+                      inPayment = false
+                    } catch (e: unknown) {
+                      inPayment = false
+                      if (
+                        e instanceof Error &&
+                        e.message === SendErrorBackPressed
+                      ) {
+                        // User canceled: do nothing
+                        return
                       }
+                      if (
+                        e instanceof Error &&
+                        e.message === SendErrorNoTransaction
+                      ) {
+                        await showUi.showToast(
+                          lstrings.fiat_plugin_sell_failed_to_send_try_again,
+                          NOT_SUCCESS_TOAST_HIDE_MS
+                        )
+                        return
+                      }
+                      await showUi.showError(e)
```

Comment:
Confirmed. Restored specific handling for user cancel (SendErrorBackPressed) and no-transaction (SendErrorNoTransaction), preserving the prior UX. Paybis still prefers segwit: getAddresses returns segwit first when present via the first element publicAddress.